crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -114,7 +111,9 @@ pub(crate) fn file_to_module(db: &dyn Db, file: File) -> Option<Module> {
 ///
 /// [module resolution order]: https://typing.readthedocs.io/en/latest/spec/distributing.html#import-resolution-ordering
 #[salsa::tracked(return_ref)]
-pub(crate) fn module_resolution_settings(db: &dyn Db) -> ModuleResolutionSettings {
+pub(crate) fn resolve_module_resolution_settings(
```

Comment:
This function now returns a `Result`. We now have another `module_resolution_settings(db: &dyn Db)` function that simply calls `resolve_module_resolution_settings(db).as_ref().unwrap()`. This still obviously isn't correct, because we're still panicking when we shouldn't, but:
- it's now isolated to one `.unwrap()` call and TODO comment
- we should get a nicer error message when the panic happens rather than just "called `.unwrap()` on a None value"